app/controllers/simulator_controller.rb

```diff
@@ -9,6 +9,8 @@ class SimulatorController < ApplicationController
   before_action :set_user_project, only: %i[update edit update_image]
   before_action :check_view_access, only: %i[show embed get_data]
   before_action :check_edit_access, only: %i[edit update update_image]
+  before_action :set_issue_circuit_data, only: %i[view_issue_circuit_data]
+  before_action :check_issue_circuit_data_access, only: %i[view_issue_circuit_data]
```

Comment:
This seems an overkill as this action is just for one method each

Reply:
Should I just include the authorization in the method itself then?

Reply:
yes